syft/messaging/message.py

```diff
@@ -195,6 +200,114 @@ def detail(worker: AbstractWorker, msg_tuple: tuple) -> "Operation":
+        protobuf_op.command = operation.cmd_name
+
+        if type(operation.cmd_owner) == sy.generic.pointers.pointer_tensor.PointerTensor:
+            protobuf_owner = protobuf_op.owner_pointer
+        else:
+            protobuf_owner = protobuf_op.owner_tensor
+        protobuf_owner.CopyFrom(sy.serde.protobuf.serde._bufferize(worker, operation.cmd_owner))
+
+        if operation.cmd_args:
+            protobuf_op.args.extend(Operation._bufferize_args(worker, operation.cmd_args))
+
+        if operation.cmd_kwargs:
+            for key, value in operation.cmd_kwargs.items():
+                protobuf_op.kwargs.get_or_create(key).CopyFrom(
+                    Operation._bufferize_arg(worker, value)
+                )
+
+        for return_id in operation.return_ids:
+            sy.serde.protobuf.proto.set_protobuf_id(protobuf_op.return_ids.add(), return_id)
+
+        protobuf_op_msg.operation.CopyFrom(protobuf_op)
+        return protobuf_op_msg
+
+    @staticmethod
+    def unbufferize(worker: AbstractWorker, protobuf_obj: "OperationMessagePB") -> "Operation":
+        """
+        This function takes the Protobuf version of this message and converts
+        it into an Operation. The bufferize() method runs the inverse of this method.
+
+        Args:
+            worker (AbstractWorker): a reference to the worker necessary for detailing. Read
+                syft/serde/serde.py for more information on why this is necessary.
+            protobuf_obj (OperationPB): the Protobuf message
+
+        Returns:
+            obj (Operation): an Operation
+
+        Examples:
+            message = unbufferize(sy.local_worker, protobuf_msg)
+        """
+
+        command = protobuf_obj.operation.command
+        owner = sy.serde.protobuf.serde._unbufferize(
+            worker, getattr(protobuf_obj.operation, protobuf_obj.operation.WhichOneof("owner"))
+        )
+        args = Operation._unbufferize_args(worker, protobuf_obj.operation.args)
+
+        kwargs = {}
+        for key in protobuf_obj.operation.kwargs:
+            kwargs[key] = Operation._unbufferize_arg(worker, protobuf_obj.operation.kwargs[key])
+
+        return_ids = [
+            sy.serde.protobuf.proto.get_protobuf_id(pb_id)
+            for pb_id in protobuf_obj.operation.return_ids
+        ]
+
+        operation_msg = Operation(command, owner, tuple(args), kwargs, tuple(return_ids))
+
+        return operation_msg
+
+    @staticmethod
+    def _bufferize_args(worker: AbstractWorker, args: list) -> list:
+        protobuf_args = []
+        for arg in args:
+            protobuf_args.append(Operation._bufferize_arg(worker, arg))
+        return protobuf_args
+
+    @staticmethod
+    def _bufferize_arg(worker: AbstractWorker, arg: object) -> ArgPB:
+        protobuf_arg = ArgPB()
+        try:
+            setattr(protobuf_arg, "arg_" + type(arg).__name__, arg)
+        except:
+            getattr(protobuf_arg, "arg_" + type(arg).__name__).CopyFrom(
```

Comment:
Hmm, what will happen if type is not present in Arg... will it just raise AttributeError twice?
Maybe explicit check for supported type (based on ArgPB) would be better?

Reply:
I think that’s what will happen, yeah. I’m not entirely sure how to do an explicit check for supported types here, but it sounds like a reasonable thing to do.